### ISSUE TITLE
setForceSpeakerphoneOn not working on Android

### ISSUE THREAD

**Issue by ndarilek** [zxcpoiu/react-native-incall-manager#3]
Using version 1.0.1. Steps to reproduce:

1. Set up a default video call where the speakerphone is on and audio is being transmitted.

2. Call `InCallManager.setForceSpeakerphoneOn(false)`.

On my phone this seems to have no effect. I'd expect this to turn off the speaker so audio is going through the phone speaker.

Thanks for the quick turnaround, 1.0.1 is working very nicely with regard to the issues I filed earlier!

**Comment by zxcpoiu** [comment]
oh, maybe is the bad naming of api
`setForceSpeakerPhoneOn()` is just a flag indicating that the `force flag` is on or off.

if you want to turn speaker on or off. you can try `setSpeakerPhoneOn(false)` and it should route audio to ear piece instead default speaker.

Maybe I should change the naming of api :/

**Comment by ndarilek** [comment]
Ah, that might be useful, yes. I'll try that later tonight.

Also, is there some way to set the value not to true or false, but to 
null and thus whatever the default would be? Like, if a user forces the 
speakerphone on but then connects a headset, they probably want audio 
through the earpiece even though the speaker was set on. I guess I'd 
like to still have the opportunity to force the speakerphone on or off 
for some use cases, but to still clear the force state and switch to the 
default value if I wanted to.

Thanks.

**Comment by zxcpoiu** [comment]
yeah, there do have lots of use cases even edge cases.
we have to address them and make a much more flexible api.

thank you for providing such a scenario.

**Comment by ndarilek** [comment]
How about this as an API:

`setSpeakerphoneOn(true)`: Set speakerphone on no matter what, don't 
update as state changes
`setSpeakerphoneOn(false)`: Set speakerphone off no matter what, don't 
update as state changes
`setSpeakerphoneOn(null)`: Set speakerphone to whatever value it should 
have for the given state, no longer force it on or off, and update it as 
state changes
Drop the `force` variant as that seems a bit confusing.

Thoughts?

**Comment by zxcpoiu** [comment]
yes, that's what I thought of, it would be much more intuitive for both user and developer to implement a `speaker phone on` button.

but the api name and flags during automatic period must be distinguishable enough.

since androd has a native api called `setSpeakerPhoneOn()` which doesn't mean `force speaker on` when wired headset plugged. that's the reason I made a misleading api name `setForceSpeakerPhoneOn()` to set a flag which indicate to force spear on even external headset plugged.

anyway, I got your point and agreed with you.

:)
